src/emqx_broker.erl

```diff
@@ -462,14 +477,30 @@ handle_cast({subscribe, Topic}, State) ->
     end,
     {noreply, State};
 
-handle_cast({unsubscribed, Topic}, State) ->
-    case ets:member(?SUBSCRIBER, Topic) of
-        false ->
-            _ = emqx_router:do_delete_route(Topic),
-            ok;
-        true -> ok
-    end,
-    {noreply, State};
+handle_cast({unsubscribed, {Topic, Caller}}, #{unsub_queue := UnSubQ,
+                                               unsub_queue_flush_timer := UnsubT
+                                              } = State) ->
+    HasPort = is_pid_links_port(Caller),
+    IsWildcard = emqx_topic:wildcard(Topic),
+    NewState = case ets:member(?SUBSCRIBER, Topic) of
+                   false when IsWildcard andalso not HasPort ->
+                       NewUnsubT = case UnsubT of
+                                       undefined ->
+                                           Wait = application:get_env(emqx, flush_unsub_timer, 60000),
+                                           {ok, TRef} = timer:send_after(Wait, flush_unsub_queue),
+                                           TRef;
+                                       OldTref ->
+                                           OldTref
+                                   end,
+                       State#{unsub_queue => [Topic | UnSubQ],
```

Comment:
Caching might get lost if broker proc get killed. will address this issue in separate PR.